opencv3: fixes opencl build error

heinemml opened the pull request:
### Have you:

- [x] Followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-science/blob/master/.github/CONTRIBUTING.md) document?
- [x] Checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-science/pulls) for the same update/change?
- [x] Run `brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting) and corrected all errors?
- [x] Built your formula locally prior to submission with `brew install <formula>`?

### Updates to existing formula: have you

- [x] Added/bumped the `revision` number if the changes affect the precompiled bottles?
- [x] Not altered the `bottle` section?
- [x] Checked that the tests still pass?

patch contains upstream commit https://github.com/Itseez/opencv/commit/c7bdbef5042dadfe032dfb5d80f9b90bec830371 which fixes https://github.com/Itseez/opencv/issues/6052
this fixes https://github.com/Homebrew/homebrew-science/issues/3147

dpo commented on opencv3.rb:
```diff
@@ -13,7 +13,9 @@ class Opencv3 < Formula
-    # this can be removed when a new version is released
+    # 2nd patch fixes build error https://github.com/Homebrew/homebrew-science/issues/3147 when not using --without-opencl
+    # fixed in https://github.com/Itseez/opencv/commit/c7bdbef5042dadfe032dfb5d80f9b90bec830371
+    # both patches can be removed when a new version is released
```
Is there any chance of grabbing the patch from `https://github.com/Itseez/opencv/commit/c7bdbef5042dadfe032dfb5d80f9b90bec830371.diff`?

heinemml replied:
Sure, didn't find a guideline on which way to prefer. I will change both patches to pull it from github.

heinemml commented:
I changed the commit to pull all patches directly from upstream.
I also fixed the style warnings issued by 'brew audit'. Is this okay or would this be better in an additional commit and/or pull request?

ahundt commented:
can this be merged soon?

dpo commented:
@BrewTestBot  test this please

heinemml commented:
From my side this PR is ready. The CI build was sucessful before but currently the brewbot seems to be in some kind of deadlock state by some changes from core...

ahundt commented:
@dpo ready to merge perhaps? Thanks!
